lib/jobs/get-catalog-values-job.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var di = require('di');
+
+module.exports = getCatalogValuesJobFactory;
+di.annotate(getCatalogValuesJobFactory, new di.Provide('Job.Get.Catalog.Values'));
+di.annotate(getCatalogValuesJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Services.Waterline',
+    'Util',
+    '_'
+));
+
+function getCatalogValuesJobFactory(
+    BaseJob,
+    Logger,
+    waterline,
+    util,
+    _
+){
+    var logger = Logger.initialize(getCatalogValuesJobFactory);
+
+    /**
+     * This job finds one or more requested values from one or more specified catalogs.
+     * The catalog values are made available in the shared context in an object that 
+     * maps the catalog values to requester provided keys.
+     *
+     * Example requestedData array:
+     * [{
+     *   "source": "ohai",
+     *   "keys": {
+     *     "myIPAddrKeyName": "data.ipaddress"
+     *   }
+     * }]
+     *
+     * Adds to context.data:
+     * {
+     *   "myIPAddrKeyName": <catalog value>
+     * }
+     *
+     * @param options
+     * @param context
+     * @param taskId
+     * @constructor
+     */
+
+    function GetCatalogValuesJob(options, context, taskId){
+        GetCatalogValuesJob.super_.call(this, logger, options, context, taskId);
+        this.nodeId = context.target;
+        this.options = options;
+        this.logger = logger;
+    }
+
+    util.inherits(GetCatalogValuesJob, BaseJob);
+
+    /**
+     * @memberOf GetCatalogValuesJob
+     */
+    GetCatalogValuesJob.prototype._run = function _run(){
+        var self = this,
+            newObj = {};
+
+        // Get the correct catalog based on source
+        _.forEach(self.options.requestedData, function(catalogInfo){
```

Comment:
@dmskinner : For only one catalog source, the log does show the sequence is OK. The key root cause is the _done is called multiple times. Actually, RackHD will immediately finish the job if `_done` is called, so that the next job will be scheduled to run. Image the next job is just want to use the fetched catalog data, the data may not be ready at that time (just check my example code of `console.log('context after forEach:');`). Of course, if there is enough time before the next job uses the context value, you may not realize there is such problem.
In summary, Make sure the `_done` is called only once is key principle for job.